packages/babel-parser/src/plugins/flow.js

```diff
@@ -1160,6 +1182,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           if (kind === "get" || kind === "set") {
             this.flowCheckGetterSetterParams(node);
           }
+          if (node.key.name === "constructor" && node.value.this) {
```

Comment:
As you can see from https://astexplorer.net/#/gist/e4e0c3380467882e445d5ddd8193feac/8bf60fc83e55d1fff70cd5b0ad4d3255e9fb5691, when `estree` plugin is enabled, the AST is slightly different. `node.value` becomes a function expression. The error is thrown, very likely because `node.value.this` is no longer `true` when `estree` is enabled.

Reply:
Can you add a new test case on string literal constructor?
```js
class A {
  "constructor"() {}
}
```
It is semantically equivalent to the identifier constructor.
```